templates/charms-architecture.html

```diff
@@ -0,0 +1,188 @@
+      </div>
+    </div>
+  </section>
+</section>
+<section class="p-section u-no-padding--bottom">
+  <section class="p-section--shallow">
+    <div class="row--50-50">
+      <div class="u-fixed-width"><hr class="p-rule"></div>
+      <div class="col">
+        <h1 class="p-heading--2">How do charms work?</h1>
+      </div>
+      <div class="col">
+        <p class="p-heading--5">A simple explanation of how charms interact with Juju and the workload they operate</p>
+        <p>The Juju controller is the brain of every deployment and it is a process that runs in the background and manages the relationships between the clients (e.g. the CLI or Terraform), the cloud substrate and the charm code. The controller also maintains the state of the system, as well as other data structures that can be accessed by the charms.</p>
+      </div>
+    </div>
+  </section>
+  <div class="row--50-50">
+    <div class="col"></div>
+    <div class="col">
+      <hr class="p-rule--muted">
+      <p class="p-heading--5">When a client issues a command the following happens:</p>
+    </div>
+  </div>
+  <div class="row--50-50">
+    <div class="col"></div>
+    <div class="col">
+      <hr class="is-muted">
+      <ol class="p-list--divided juju-middle-pages">
+        <li class="p-list__item">When the client parses your command it sends a request to the Juju controller API.</li>
+        <li class="p-list__item">The Juju controller parses the command and makes a change to the data structure representing the model, as well as the cloud substrate to represent what the command says (e.g. adding a unit).</li>
+        <li class="p-list__item">The Juju agents running on the charm constantly monitor the controller, looking for changes that affect them. When there is a discrepancy between their local state and the one represented in the controller they compute a diff and create some “events”.</li>
+        <li class="p-list__item">The agent then executes the charm code, which includes a specific set of actions that need to occur when a specific event occurs, as well as logic to specify their relative ordering.</li>
+        <li class="p-list__item">The charm operates the workload in whatever way appropriate to the event being handled; if necessary, it will interact with the live workload through pebble, to read/write the workload filesystem, run commands, etc…</li>
+        <li class="p-list__item">The remote state is aligned with the one represented in the controller.</li>
+      </ol>
+    </div>
+  </div>
+  <div class="row--50-50">
+    <div class="col"></div>
+    <div class="col">
+      <p>For more detailed information you can read our <a href="/docs/sdk/charm-lifecycle">charm lifecycle documentation article&nbsp;&gt;</a></p>
+    </div>
+  </div>
+  <div class="row">
+    <div class="col-6 col-start-large-7" style="padding-top: 2rem;">
```

Comment:
I don't know if this has to be 2rem specifically by design, but according to Vanilla ideally a section component should be used to provide padding bottom.

In this case, a row above would be wrapped in shallow section to provide 1.5rem padding bottom.